crates/sui-json-rpc-types/src/sui_transaction.rs

```diff
@@ -956,6 +957,26 @@ pub struct SuiTransactionBlockEvents {
 
 impl SuiTransactionBlockEvents {
     pub fn try_from(
+        events: TransactionEvents,
+        tx_digest: TransactionDigest,
+        timestamp_ms: Option<u64>,
+        resolver: &mut dyn LayoutResolver,
+    ) -> SuiResult<Self> {
+        Ok(Self {
+            data: events
+                .data
+                .into_iter()
+                .enumerate()
+                .map(|(seq, event)| {
+                    let layout = resolver.get_annotated_layout(&event.type_)?;
+                    SuiEvent::try_from(event, tx_digest, seq as u64, timestamp_ms, layout)
+                })
+                .collect::<Result<_, _>>()?,
+        })
+    }
+
+    // TODO: this is only called from the indexer. Remove this once indexer moves to its own resolver.
```

Comment:
:+1:

I was going to ask whether we had any remaining uses of this API, or the `ModuleCache` or `GetModule` -- if not, we should get rid of them entirely to avoid temptation (not blocking for this PR).